lib/manager/pip_setup/extract.ts

```diff
@@ -1,139 +1,79 @@
-      const version = parsePythonVersion(stdout || stderr);
-      if (version[0] >= 3 && version[1] >= 7) {
-        pythonAlias = pythonVersion;
-        break;
-      }
-    } catch (err) {
-      logger.debug(`${pythonVersion} alias not found`);
-    }
-  }
-  return pythonAlias;
+function cleanupNamedGroups(regexSource: string): string {
+  return regexSource.replace(/\(\?<\w+>/g, '(?:');
 }
 
-export async function extractSetupFile(
-  _content: string,
-  packageFile: string,
-  config: ExtractConfig
-): Promise<PythonSetup> {
-  let cmd = 'python';
-  const extractPy = await getExtractFile();
-  const args = [`"${extractPy}"`, `"${packageFile}"`];
-  if (getGlobalConfig().binarySource !== 'docker') {
-    logger.debug('Running python via global command');
-    cmd = await getPythonAlias();
-  }
-  logger.debug({ cmd, args }, 'python command');
-  const res = await exec(`${cmd} ${args.join(' ')}`, {
-    cwdFile: packageFile,
-    timeout: 30000,
-    docker: {
-      image: 'python',
+const rangePattern = cleanupNamedGroups(RANGE_PATTERN);
+const versionPattern = `(?:${rangePattern}(?:\\s*,\\s*${rangePattern})*)`;
+const depNamePattern = '(?:[a-zA-Z][-_a-zA-Z0-9]*[a-zA-Z0-9])';
+const depPattern = [
+  '^',
+  `(?<depName>${depNamePattern})`,
+  `(?<extra>(?:\\[\\s*(?:${depNamePattern}(?:\\s*,\\s*${depNamePattern})*\\s*)\\])?)`,
+  `(?<currentValue>${versionPattern})`,
+].join('\\s*');
+
+const extractRegex = new RegExp(depPattern);
+
+function depStringHandler(
+  ctx: Context,
+  token: lexer.StringValueToken
+): Context {
+  const depStr = token.value;
+  const match = extractRegex.exec(depStr);
+  const { depName, currentValue } = match.groups;
+
+  const dep: PackageDependency<ManagerData> = {
+    depName,
+    currentValue,
+    managerData: {
+      lineNumber: token.line - 1,
     },
-  });
-  if (res.stderr) {
-    const stderr = res.stderr
-      .replace(/.*\n\s*import imp/, '')
-      .trim()
-      .replace('fatal: No names found, cannot describe anything.', '');
-    if (stderr.length) {
-      logger.warn({ stdout: res.stdout, stderr }, 'Error in read setup file');
-    }
-  }
-  return parseReport(packageFile);
+    datasource: PypiDatasource.id,
+  };
+
+  return { ...ctx, deps: [...ctx.deps, dep] };
+}
+
+function depSkipHandler(ctx: Context): Context {
+  const dep = ctx.deps[ctx.deps.length - 1];
+  const deps = ctx.deps.slice(0, -1);
+  deps.push({ ...dep, skipReason: SkipReason.Ignored });
+  return { ...ctx, deps };
 }
 
-export async function extractPackageFile(
+const incompleteDepString = q
+  .str<Context>(new RegExp(cleanupNamedGroups(depPattern)))
+  .op(/^\+|\*$/);
+
+const depString = q
+  .str<Context>(new RegExp(cleanupNamedGroups(depPattern)), depStringHandler)
+  .opt(
+    q
+      .opt(q.op<Context>(','))
+      .comment(/^#\s*renovate\s*:\s*ignore\s*$/, depSkipHandler)
+  );
+
+const query = q.alt(incompleteDepString, depString);
```

Comment:
I'm trying to think the best way to document everything from here on in.. in case you get hit by a bus tomorrow.

Reply:
Yeah, I definitely need to document this API. Though, let's not hypothesise on my death too much 😅

Reply:
I was meaning that your laptop, password manager and 2FA were all destroyed.. you're fine of course :)

Reply:
Understand, I'm just kidding :)